### ISSUE TITLE
PVC fails when using uid and gid in storage class  

### ISSUE THREAD

**Issue by przem123** [IBM/ibm-spectrum-scale-csi#60]
For my storage class I used `uid` and `gid` params as stated in the Knowledge Center.

```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
   name: ibm-spectrum-scale-csi-remotefs-gid
provisioner: spectrumscale.csi.ibm.com
parameters:
    volBackendFs: "remote_fs0_1m"
    clusterId: "4473793006880872527"
    uid: "3000"
    gid: "3000"
reclaimPolicy: Delete
```

However, when I create a PVC with this class, it fails with an error `[Remote call completed with error [400 Bad Request]]`. I can see these messages in the logs:

```
I0129 20:29:09.676230       1 rest_v2.go:64] rest_v2 isRequestAccepted. url: https://10.10.1.52:443/scalemgmt/v2/filesystems/fs0_1m/filesets, response: {{202 The request was accepted for processing.} [{{[] [] 0 [] []} {POST /scalemgmt/v2/filesystems/fs0_1m/filesets map[comment:Fileset created by IBM Container Storage Interface driver filesetName:pvc-00e5a5f3-42d6-11ea-b63b-7ed30a6ab53f inodeSpace:new owner:3000:3000]} 3000000000126 2020-01-29 21:29:09,672 N/A RUNNING}]}
I0129 20:29:09.676299       1 rest_v2.go:44] rest_v2 isStatusOK. statusCode: 202
I0129 20:29:09.676313       1 rest_v2.go:77] rest_v2 waitForJobCompletion. jobID: 3000000000126, statusCode: 202
I0129 20:29:09.676324       1 rest_v2.go:55] rest_v2 checkAsynchronousJob. statusCode: 202
I0129 20:29:09.676336       1 http_utils.go:60] http_utils FormatURL. url: https://10.10.1.52:443/
I0129 20:29:09.676351       1 rest_v2.go:90] rest_v2 AsyncJobCompletion. jobURL: https://10.10.1.52:443/scalemgmt/v2/jobs/3000000000126?fields=:all:
I0129 20:29:09.676363       1 rest_v2.go:534] rest_v2 doHTTP. endpoint: https://10.10.1.52:443/scalemgmt/v2/jobs/3000000000126?fields=:all:, method: GET, param: <nil>
I0129 20:29:09.676378       1 http_utils.go:74] http_utils HttpExecuteUserAuth. type: GET, url: https://10.10.1.52:443/scalemgmt/v2/jobs/3000000000126?fields=:all:, user: csiadmin
I0129 20:29:09.704182       1 http_utils.go:44] http_utils UnmarshalResponse. response: &{0xc0001cc4c0 {0 0} false <nil> 0x6e1a80 0x6e1a10}
I0129 20:29:09.704308       1 rest_v2.go:44] rest_v2 isStatusOK. statusCode: 200
I0129 20:29:11.704552       1 rest_v2.go:534] rest_v2 doHTTP. endpoint: https://10.10.1.52:443/scalemgmt/v2/jobs/3000000000126?fields=:all:, method: GET, param: <nil>
I0129 20:29:11.704595       1 http_utils.go:74] http_utils HttpExecuteUserAuth. type: GET, url: https://10.10.1.52:443/scalemgmt/v2/jobs/3000000000126?fields=:all:, user: csiadmin
I0129 20:29:11.736244       1 http_utils.go:44] http_utils UnmarshalResponse. response: &{0xc0001cc640 {0 0} false <nil> 0x6e1a80 0x6e1a10}
I0129 20:29:11.736455       1 rest_v2.go:44] rest_v2 isStatusOK. statusCode: 200
E0129 20:29:11.736475       1 rest_v2.go:111] Async Job failed: {{200 The request finished successfully.} [{{[] [] 9 [EFSSP0010C CLI parser: The object "3000" specified for "user" does not exist.] []} {POST /scalemgmt/v2/filesystems/fs0_1m/filesets map[comment:Fileset created by IBM Container Storage Interface driver filesetName:pvc-00e5a5f3-42d6-11ea-b63b-7ed30a6ab53f inodeSpace:new owner:3000:3000]} 3000000000126 2020-01-29 21:29:09,672 2020-01-29 21:29:09,705 FAILED}]}
E0129 20:29:11.736536       1 rest_v2.go:295] Unable to create fileset pvc-00e5a5f3-42d6-11ea-b63b-7ed30a6ab53f: [EFSSP0010C CLI parser: The object "3000" specified for "user" does not exist.]
I0129 20:29:11.736553       1 rest_v2.go:390] rest_v2 ListFileset. filesystem: fs0_1m, fileset: pvc-00e5a5f3-42d6-11ea-b63b-7ed30a6ab53f
I0129 20:29:11.736566       1 http_utils.go:60] http_utils FormatURL. url: https://10.10.1.52:443/
I0129 20:29:11.736582       1 rest_v2.go:534] rest_v2 doHTTP. endpoint: https://10.10.1.52:443/scalemgmt/v2/filesystems/fs0_1m/filesets/pvc-00e5a5f3-42d6-11ea-b63b-7ed30a6ab53f, method: GET, param: <nil>
I0129 20:29:11.736594       1 http_utils.go:74] http_utils HttpExecuteUserAuth. type: GET, url: https://10.10.1.52:443/scalemgmt/v2/filesystems/fs0_1m/filesets/pvc-00e5a5f3-42d6-11ea-b63b-7ed30a6ab53f, user: csiadmin
I0129 20:29:11.763780       1 http_utils.go:44] http_utils UnmarshalResponse. response: &{0xc00025aa40 {0 0} false <nil> 0x6e1a80 0x6e1a10}
I0129 20:29:11.763959       1 rest_v2.go:44] rest_v2 isStatusOK. statusCode: 400
E0129 20:29:11.763995       1 rest_v2.go:397] Error in list fileset request: Remote call completed with error [400 Bad Request]
E0129 20:29:11.764019       1 utils.go:55] GRPC error: rpc error: code = Internal desc = Unable to create fileset [pvc-00e5a5f3-42d6-11ea-b63b-7ed30a6ab53f] in FS [fs0_1m]. Error [Remote call completed with error [400 Bad Request]]
```

Please provide a fix for this problem.

**Comment by przem123** [comment]
Can't you remove this restriction ?  With remote clusters, the existence of the user cannot be granted as the system may be managed by another admin.

**Comment by smitaraut** [comment]
@przem123 no this restriction cannot be removed until we integrate with some authentication server. For now we are adding this pre-req to the IBM knowledge center documentation.

**Comment by deeghuge** [comment]
Fixed in Spectrum Scale 5.0.5.2 and 5.1.0.
